base/strings/types.jl

```diff
@@ -117,10 +117,23 @@ end
 
 Reverses a string.
 
+Technically, this function reverses the "codepoints" in a string, and its
+main utility is for reversed-order string processing, especially for reversed
+regex (regular expression) searches.  See also [`reverseind`](@ref) to convert indices
+in `s` to indices in `reverse(s)` and vice-versa, and [`graphemes`](@ref)
+to operate on user-visible "characters" (graphemes) rather than codepoints
+(which is more [visually pleasing but less useful](https://github.com/JuliaLang/julia/issues/6165) for reverse searching).
+
 # Examples
 ```jldoctest
 julia> reverse("JuliaLang")
 "gnaLailuJ"
+
+julia> reverse("ax̂e") # combining characters can lead to surprising results
+"êxa"
+
+julia> join(reverse(collect(graphemes("ax̂e")))) # reverses graphemes
+"ex̂a"
```

Comment:
The hat is shown in the wrong place in github's diff display in by browser, but if I paste it into my terminal it is correct.  Apparently their display of combining characters is broken?

![image](https://user-images.githubusercontent.com/2913679/30656314-afbe4c76-9e01-11e7-983f-e57281ea5738.png)